File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {

        // remove non-alphanumeric characters
        s = regex_replace(s,regex("[^a-zA-Z0-9]"),"");

        // initialize pointers
        int start = 0, end = s.size()-1;

        // run loop while pointers point to different elements
        while(end > start){
            // if(pointer does not point to equal alphanumeric characters, return false)
            if(tolower(s[start]) != tolower(s[end])) return false;
            start++;
            end--;
        }

        return true;

    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        // initialize pointers
        let start=0, end=numbers.length-1;

        // run loop while start pointer < end pointer
        while(start < end){
            // if sum = target return index array
            if(numbers[start]+numbers[end] == target) return [start+1,end+1];
            // else if sum > target, end = end-1
            else if(numbers[start]+numbers[end] > target) end--;
            // else (sum < target), start = start+1
            else start++;
        }
    }
}
